src/Suggestor/Suggestor.ts

```diff
@@ -464,10 +464,10 @@ function addDependsOnSuggestions(
  * Matches a string with a regex according to a position (typically of a cursor).
  * Will return a result only if a match exists and the given position is part of it.
  */
-export function matchByPosition(s: string, r: RegExp, position: number): RegExpMatchArray | void {
+export function matchIfCursorInRegex(s: string, r: RegExp, position: number): RegExpMatchArray | void {
     const matches = s.matchAll(r);
     for (const match of matches) {
-        if (match?.index && match.index <= position && position <= match.index + match[0].length) return match;
+        if (match?.index && match.index < position && position <= match.index + match[0].length) return match;
```

Comment:
This Change was necessary, as it Returned a Match for the Next Symbol when the Cursor was right at/before the Symbol.
Therefore it would Suggest Items for the Next Symbol, instead of the One the Symbol currently is at.

For further Explanaition see (Carriage Position Marked in Red):
![Screenshot (4)](https://github.com/obsidian-tasks-group/obsidian-tasks/assets/16257607/e57a8241-5bed-4213-b388-4c042d9531b7)

After the Fix:
![Screenshot (3)](https://github.com/obsidian-tasks-group/obsidian-tasks/assets/16257607/a9447746-abe4-4a06-bb0c-fb4fd89d881a)

I Refractored the Name in the Hope of Conveying more Clearly what the Function does.

Reply:
Thanks for improving the function name.